packages/services/f-braze-adapter/src/services/BrazeConsumerRegistry.js

```diff
@@ -0,0 +1,113 @@
+import BrazeConsumer from './BrazeConsumer';
+import BrazeDispatcher from './BrazeDispatcher';
+import DispatcherEventStream from './DispatcherEventStream';
+import { CONTENT_CARDS_EVENT_NAME, IN_APP_MESSAGE_EVENT_CLICKS_NAME, IN_APP_MESSAGE_EVENT_NAME } from './types/events';
+
+let registryInstance;
+
+/**
+ * sessionTimeoutInSeconds
+ * Set session timeout to 0 in order to avoid caching issues with Braze
+ * @type {number}
+ */
+export const sessionTimeoutInSeconds = 0;
+
+class BrazeConsumerRegistry {
+    /**
+     * Static constructor to store one instance of BrazeDispatcher per js env
+     * @return {BrazeConsumerRegistry}
+     * @constructor
+     */
+    static GetConsumerRegistry (options) {
+        if (!registryInstance) {
+            registryInstance = new BrazeConsumerRegistry(options);
+        }
+        return registryInstance;
+    }
+
+    get dispatcher () {
+        return this._dispatcher;
+    }
+
+    constructor ({
+        apiKey,
+        userId,
+        enableLogging,
+        sessionTimeout = sessionTimeoutInSeconds
+    }) {
+        this.consumers = [];
+
+        this.eventStream = new DispatcherEventStream();
+
+        this._dispatcher = new BrazeDispatcher(sessionTimeout, this.eventStream);
+
+        // TODO change this so we call all configure bits inside the dispatcher constructor
+        this._dispatcher.configure({
+            apiKey,
+            userId,
+            enableLogging
+        });
```

Comment:
Note configure is `async` at this point, so we either need to `await` it (and catch any errors/rejections) - or attach a `.catch()` to the resulting promise, to avoid having the potential for an unhandled promise rejection.

Reply:
resolved now by using the new BrazeAdapter initialize static function, which returns the braze adapter instance.